### PR TITLE
move realtime and SEV related nodeSelectors from VMI to the vmipod

### PR DESCRIPTION
**Pull request by vladikr**
### What this PR does
This PR moves the setting of certain nodeSelectors from the VMI admonition controller to virt-controller.
nodeSelectors affect the scheduling of the virt-launcher pod and do not affect the VMI itself. 
Therefore, these should be combined with the rest of the nodeSelector generators in the virt-controller and be set directly on the virt-launcher pod.

Before this PR:
The SEV(-ES) and realtime nodeSelectors were added to the VMI at admission.

After this PR:
The above nodeSelectors will be added directly to the virt-launcher pod by the virt-controller.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/services/template_test.go [hunk 1]
```diff
@@ -1123,6 +1123,114 @@ var _ = Describe("Template", func() {
...
+						SEV: &v1.SEV{
+							Policy: &v1.SEVPolicy{
+								EncryptedState: pointer.P(false),
+							},
+						},
+					}
+					pod, err := svc.RenderLaunchManifest(vmi)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(pod.Spec.NodeSelector).To(Not(HaveKey(ContainSubstring(v1.SEVESLabel))))
+				})
```
Comment (fossedihelm) [inline pkg/virt-controller/services/template_test.go:1231]: nit:
```suggestion
				DescribeTable("should not add SEV-ES node label selector", func(policy *v1.SEVPolicy) {
					vmi.Spec.Domain.LaunchSecurity = &v1.LaunchSecurity{
						SEV: &v1.SEV{
							Policy: policy,
						},
					}

					pod, err := svc.RenderLaunchManifest(vmi)
					Expect(err).ToNot(HaveOccurred())
					Expect(pod.Spec.NodeSelector).To(Not(HaveKey(ContainSubstring(v1.SEVESLabel))))

				},
					Entry("when no SEV policy is set", &v1.SEVPolicy{}),
					Entry("when no SEV-ES policy bit is set", &v1.SEVPolicy{EncryptedState: nil}),
					Entry("when SEV-ES policy bit is set to false", &v1.SEVPolicy{EncryptedState: pointer.P(false)}),
				)
```

**Review by fossedihelm — COMMENTED** [review]
Changes look reasonable and the code looks good.
Just a nit, but feel free to ignore it.
Maybe it's worth it to add something to the release note, informing that the VMI node selector does not show Realtime/SEV selector anymore. WDYT?

Also, can you drop the `addNodeSelector` function in vmi-mutator file?
https://github.com/kubevirt/kubevirt/blob/a16cc70070333fc33e40c9bfcd60f94c3f7b6c06/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L161-L166

Thank you!

File: pkg/virt-controller/services/template_test.go (hunk 1 above)
Comment (vladikr) [reply pkg/virt-controller/services/template_test.go:1231]: Yes, thanks :)

**Comment by vladikr** [comment]
@fossedihelm Thanks for the review!
wrt: 

> "Maybe it's worth it to add something to the release note, informing that the VMI node selector does not show Realtime/SEV selector anymore. WDYT?" 

I wouldn't, these nodelectors had no meaning for the VMI and didn't need to be there in the first place.

**Review by fossedihelm — APPROVED** [review]
Thank you!

**Comment by fossedihelm** [comment]
> I wouldn't, these nodelectors had no meaning for the VMI and didn't need to be there in the first place.

Agreed. My only concern was that some vendors could, don't know why, rely on those.

**Review by jean-edouard — COMMENTED** [review]
/approve

**Comment by vladikr** [comment]
> > I wouldn't, these nodelectors had no meaning for the VMI and didn't need to be there in the first place.
> 
> Agreed. My only concern was that some vendors could, don't know why, rely on those.

That would be unexpected since both of these nodeselectors are being derived from a VMI API field.

**Pull request merged**